contracts/Safe.sol

```diff
@@ -24,7 +24,9 @@ import {ISafe} from "./interfaces/ISafe.sol";
  *      - Transaction Hash: Hash of a transaction is calculated using the EIP-712 typed structured data hashing scheme.
  *      - Nonce: Each transaction should have a different nonce to prevent replay attacks.
  *      - Signature: A valid signature of an owner of the Safe for a transaction hash.
- *      - Guard: Guard is a contract that can execute pre- and post- transaction checks. Managed in `GuardManager`.
+ *      - Guards: Guards are contracts that can execute pre- and post- transaction checks. There are two types of guards:
+ *          1. Transaction Guard - Managed in `GuardManager`.
+ *          2. Module Guard - Managed in `ModuleManager`.
```

Comment:
Nice! I really like the nat-spec improvements. I think it would be nice to further specify what each of the guard types do:

```suggestion
 *          1. Transaction Guard: managed in `GuardManager` for transactions executed with `execTransaction`.
 *          2. Module Guard: managed in `ModuleManager` for transactions executed with `execTransactionFromModule`
```